From 015ac790dd84677e121c9da1d3c2c32d95a6d913 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:07:11 -0400
Subject: [powerpc] kernel: remove useless code which related with 'max_cpus'

Message-id: <1410545655-205645-203-git-send-email-dzickus@redhat.com>
Patchwork-id: 94278
O-Subject: [RHEL7 PATCH 202/626] powerpc: kernel: remove useless code which related with 'max_cpus'
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit dfee0efe3ec8d4099c69e8234e4e4306619b9ba6
Author: Chen Gang <gang.chen@asianux.com>
Date:   Mon Jul 22 14:40:20 2013 +0800

    powerpc: kernel: remove useless code which related with 'max_cpus'

    Since not need 'max_cpus' after the related commit, the related code
    are useless too, need be removed.

    The related commit:

      c1aa687 powerpc: Clean up obsolete code relating to decrementer and timebase

    The related warning:

      arch/powerpc/kernel/smp.c:323:43: warning: parameter 'max_cpus' set but not used [-Wunused-but-set-parameter]

    Signed-off-by: Chen Gang <gang.chen@asianux.com>
    Reviewed-by: Srivatsa S. Bhat <srivatsa.bhat@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/smp.c b/arch/powerpc/kernel/smp.c
index c1cf4a1..ac2621a 100644
--- a/arch/powerpc/kernel/smp.c
+++ b/arch/powerpc/kernel/smp.c
@@ -369,13 +369,8 @@ void __init smp_prepare_cpus(unsigned int max_cpus)
  cpumask_set_cpu(boot_cpuid, cpu_sibling_mask(boot_cpuid));
  cpumask_set_cpu(boot_cpuid, cpu_core_mask(boot_cpuid));
 
- if (smp_ops)
-  if (smp_ops->probe)
-   max_cpus = smp_ops->probe();
-  else
-   max_cpus = NR_CPUS;
- else
-  max_cpus = 1;
+ if (smp_ops && smp_ops->probe)
+  smp_ops->probe();
 }
 
 void smp_prepare_boot_cpu(void)
-- 
1.7.1